x/incentives/keeper/gauge.go

```diff
@@ -546,3 +546,57 @@ func (k Keeper) initGaugeInfo(ctx sdk.Context, poolIds []uint64) (types.Internal
 		GaugeRecords: gaugeRecords,
 	}, nil
 }
+
+// GetPoolIdsAndDurationsFromGroup retrieves the pool IDs and their associated durations from a given group.
+// It iterates over the gauge records in the group's InternalGaugeInfo, and for each record, it retrieves the pool ID and duration.
+// The function returns two slices: one for the pool IDs and one for the durations. The indices in these slices correspond to each other.
+// If there is an error retrieving the pool ID and duration for any gauge record, the function returns an error.
+func (k Keeper) GetPoolIdsAndDurationsFromGroup(ctx sdk.Context, group types.Group) ([]uint64, []time.Duration, error) {
+	poolIds := make([]uint64, 0, len(group.InternalGaugeInfo.GaugeRecords))
+	durations := make([]time.Duration, 0, len(group.InternalGaugeInfo.GaugeRecords))
+	for _, gaugeRecord := range group.InternalGaugeInfo.GaugeRecords {
+		poolId, gaugeDuration, err := k.GetPoolIdAndDurationFromGaugeRecord(ctx, gaugeRecord)
+		if err != nil {
+			return nil, nil, err
+		}
+		poolIds = append(poolIds, poolId)
+		durations = append(durations, gaugeDuration)
+	}
+	return poolIds, durations, nil
+}
+
+// GetPoolIdAndDurationFromGaugeRecord retrieves the pool ID and duration associated with a given gauge record.
+// The function first retrieves the gauge associated with the gauge record.
+// If the gauge's lock query type is NoLock, the function sets the gauge duration to the epoch duration.
+// Otherwise, it sets the gauge duration to the longest lockable duration.
+// The function then retrieves the pool ID associated with the gauge ID and the gauge duration.
+// The function returns the pool ID and the gauge duration.
+// If there is an error retrieving the gauge, the longest lockable duration, or the pool ID, the function returns an error.
+func (k Keeper) GetPoolIdAndDurationFromGaugeRecord(ctx sdk.Context, gaugeRecord types.InternalGaugeRecord) (uint64, time.Duration, error) {
+	gauge, err := k.GetGaugeByID(ctx, gaugeRecord.GaugeId)
+	if err != nil {
+		return 0, 0, err
+	}
+	gaugeType := gauge.DistributeTo.LockQueryType
+	gaugeDuration := time.Duration(0)
+
+	if gaugeType == lockuptypes.NoLock {
+		// If NoLock, it's a CL pool, so we set the "lockableDuration" to epoch duration
+		gaugeDuration = k.GetEpochInfo(ctx).Duration
+	} else {
+		// Otherwise, it's a balancer pool so we set it to longest lockable duration
+		// TODO: add support for CW pools once there's clarity around default gauge type.
+		// Tracked in issue https://github.com/osmosis-labs/osmosis/issues/6403
+		gaugeDuration, err = k.pik.GetLongestLockableDuration(ctx)
+		if err != nil {
+			return 0, 0, err
+		}
+	}
```

Comment:
I know that this logic existed before but it would be great to extend this to
```
else if gaugeType == lockuptypes.ByDuration {
    // ...
} else {
    // error here
}
```

Reply:
yeah good call https://github.com/osmosis-labs/osmosis/actions/runs/6358732266/job/17271296171?pr=6601